analysis/estimate_trfs_reference_strategy.py

```diff
@@ -0,0 +1,86 @@
+"""This script estimates TRFs for several models and saves them"""
+from pathlib import Path
+import re
+
+import eelbrain
+import mne
+import trftools
+import numpy as np
+
+
+STIMULI = [str(i) for i in range(1, 13)]
+DATA_ROOT = Path("~").expanduser() / 'Data' / 'Alice'
+PREDICTOR_DIR = DATA_ROOT / 'predictors'
+EEG_DIR = DATA_ROOT / 'eeg'
+SUBJECTS = [path.name for path in EEG_DIR.iterdir() if re.match(r'S\d*', path.name)]
+# Define a target directory for TRF estimates and make sure the directory is created
+TRF_DIR = DATA_ROOT / 'TRFs'
+TRF_DIR.mkdir(exist_ok=True)
+
+# Load stimuli
+# ------------
+# Load the broad-band envelope 
+envelope = [eelbrain.load.unpickle(PREDICTOR_DIR / f'{stimulus}~gammatone-1.pickle') for stimulus in STIMULI]
+envelope = [x.bin(0.01, dim='time', label='start') for x in envelope]
+envelope = [trftools.pad(x, tstart=-0.100, tstop=x.time.tstop + 1, name='envelope') for x in envelope]
+
+# Extract the duration of the stimuli, so we can later match the EEG to the stimuli
+durations = [gt.time.tmax for stimulus, gt in zip(STIMULI, envelope)]
+
+# Models
+# ------
+# Pre-define models here to have easier access during estimation. In the future, additional models could be added here and the script re-run to generate additional TRFs.
+models = {
+    # Acoustic models
+    'envelope': [envelope],
+}
+
+# Estimate TRFs
+# -------------
+# Loop through subjects to estimate TRFs
+for subject in SUBJECTS:
+    subject_trf_dir = TRF_DIR / subject
+    subject_trf_dir.mkdir(exist_ok=True)
+
+    for reference in ['cz', 'average']:
+        # Generate all TRF paths so we can check whether any new TRFs need to be estimated
+        trf_paths = {model: subject_trf_dir / f'{subject} {model}_{reference}.pickle' for model in models}
+        # Skip this subject if all files already exist
+        if all(path.exists() for path in trf_paths.values()):
+            continue
+        # Load the EEG data
+        raw = mne.io.read_raw(EEG_DIR / subject / f'{subject}_alice-raw_{reference}.fif', preload=True)
```

Comment:
This fails, have you tried running your script?

Reply:
On some more thought I'm assuming you saved raw data with different references on your end and that's why you did not notice this. You'll need to re-reference the raw data here.

Reply:
done